### ISSUE TITLE
vault oidc role mapping

### ISSUE THREAD

**Issue by agill17** [hashicorp/vault-plugin-auth-jwt#52]
Hi I am new to vault and OIDC in general so sorry for newbie questions 
I am trying to setup oidc with dex. 
So my config for oidc looks like this;

```
vault write auth/oidc/config \
  oidc_discovery_url="my-dex-url" \
  oidc_client_id="my-client-id" \
  oidc_client_secret="my-super-secret" 
```

and then I create a role `foo` with these parameters ;
```
    "user_claim": "sub",
    "groups_claim": "groups",
    "oidc_scopes": ["groups"],
    "bound_claims": {
        "groups": [ "foo"]
    },
}
```

Then I drop that role in `auth/oidc/role/foo`

So here is what I am not understanding and this is most likely due to the fact that I am not aware of how to wire things up yet,

When I try to login using UI, or CLI, I have to provide a role to login as
  - Is there a way to map roles based on what groups dex is returning?
If not, how do I make sure that not anyone can login to any role ? for example if I am in foo group, then I should be able to login using foo role ONLY. Is there documentation on how I can configure that mapping?

**Comment by kalafut** [comment]
Hi. Some comments:

>Is there a way to map roles based on what groups dex is returning?

There is no relationship between the role name and anything, really. It is just a name for the container of various bindings, policies, etc., and must be provided by whoever is logging in. All control of determining whether the person can login, and what permissions they have, is based on the contents of the role definition. i.e. you could rename your role to "bar" and it would still allow only "foo" group members based on the bound_claims you've set up.

>Iow do I make sure that not anyone can login to any role

bound_* parameters. You're correctly using bound_claims to limit access to only those belonging to the foo group.

BTW, this sort of question if best for the newsgroup (https://groups.google.com/forum/#!forum/vault-tool) as you'll have a wider audience both for getting answers and teaching others.

**Comment by agill17** [comment]
@kalafut Thanks for the explanation. However when using bound_claims parameters in my foo group, and I try logging in through the UI I get this;
```
error validating claims: claim "groups" does not match any associated bound claim values
```

my groups from dex does return a list, and this seems to be related to this PR; https://github.com/hashicorp/vault-plugin-auth-jwt/pull/50 which also looks like it has not been released yet to the main upstream vault repo ?

**Comment by kalafut** [comment]
Yes, if you're receiving a list of values you'll need that update which isn't bundled into Vault yet. You can build and register the latest version of plugin yourself if you want access to the feature now.

**Comment by samvrocks** [comment]
> Yes, if you're receiving a list of values you'll need that update which isn't bundled into Vault yet. You can build and register the latest version of plugin yourself if you want access to the feature now.

Will it be included in 1.1.4 release?

**Comment by kalafut** [comment]
@samvrocks Sorry for the delay... missed your comment. This change will be in 1.2 (and is in the betas now), but not the 1.1.x releases. For those older versions you'd have to mount it as an external plugin.
